chore: update refactor docs (#145)
canonical/canonicalwebteam.store-api: refactor_docs/overview.md

- Move all endpoints hitting api.snapcraft.io to `devicegw.py`
- Refactor Charmhub and snapcraft to use the refcatored module
- Document all changes made

## How methods are reorganized

> In the past, we have the `Publisher` and `Store` classes that were inherited by `CharmPublisher`, `SnapPublisher` and `CharmStore`, `SnapStore` respectively. These groupings were based on the store frontends (Charmhub and Snapcraft). Endpoints related to publisher features in `Charmhub` and `Snapcraft` were found in `CharmPublisher`  and `SnapPublisher` respectively, while endpoints related to stores in `Charmhub` and `Snapcraft` were found in `CharmStore` and `SnapStore` respectively. These classes (CharmPublisher, CharmStore, SnapPublisher, SnapStore) all consumes API endpoints from one or more of the three available API gateways which are publisher-gateway(https://api.charmhub.io), dashboard-SCA(https://dashboard.snapcraft.io), device-gateway(https://api.snapcraft.io).

Reviewer: Small edit suggestions

```suggestion
In the past, we had the `Publisher` and `Store` classes that were inherited by `CharmPublisher`, `SnapPublisher` and `CharmStore`, `SnapStore` respectively. These groupings were based on the store frontends (Charmhub and Snapcraft). Endpoints related to publisher features in `Charmhub` and `Snapcraft` were found in `CharmPublisher`  and `SnapPublisher` respectively, while endpoints related to stores in `Charmhub` and `Snapcraft` were found in `CharmStore` and `SnapStore` respectively. These classes (`CharmPublisher`, `CharmStore`, `SnapPublisher`, `SnapStore`) all consume API endpoints from one or more of the three available API gateways which are [publisher-gateway](https://api.charmhub.io), [dashboard-SCA](https://dashboard.snapcraft.io), [device-gateway](https://api.snapcraft.io).
```

> In the past, we have the `Publisher` and `Store` classes that were inherited by `CharmPublisher`, `SnapPublisher` and `CharmStore`, `SnapStore` respectively. These groupings were based on the store frontends (Charmhub and Snapcraft). Endpoints related to publisher features in `Charmhub` and `Snapcraft` were found in `CharmPublisher`  and `SnapPublisher` respectively, while endpoints related to stores in `Charmhub` and `Snapcraft` were found in `CharmStore` and `SnapStore` respectively. These classes (CharmPublisher, CharmStore, SnapPublisher, SnapStore) all consumes API endpoints from one or more of the three available API gateways which are publisher-gateway(https://api.charmhub.io), dashboard-SCA(https://dashboard.snapcraft.io), device-gateway(https://api.snapcraft.io).

Reviewer 2: Do we want people to be able to see the documentation related to each API gateway? If so, maybe we should link to `/docs` for PGW and SCA, and `/docs/index.html` for DeviceGW, as it's not super clear where the docs are if you don't know!
